src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+using ProgressMeter
+"""
+    t,y = ode_ms(odefun, y0, tspan, order=4) with tspan = [t0:dt:tfinal]
+
+Fixed-Equal Steps Adams-Bashforth multi-step solver for nonstiff problems
+
+(Main ref: Hairer & Wanner 1996, Vol I, p.359-360, 372)
+
+Note: If method is run at order k, the first step is taken at order 1,
+and the order increases each step until it is order k on the kth step.
+Thus results may be of lower accuracy than expected.
+"""
+
+function ode_ms(F, y0, tspan, order::Integer)
+    dt = diff(tspan)
+
+    y = Array(typeof(y0), length(tspan))
+    y[1] = y0
+
+    if 1 <= order <= 4
+        b = ms_coefficients4
+    else
+        b = zeros(order, order)
+        b[1:4, 1:4] = ms_coefficients4
+        for s = 5:order
+            for j = 0:(s - 1)
+                # Assign in correct order for multiplication below
+                #  (a factor depending on j and s) .* (an integral of a polynomial with -(0:s), except -j, as roots)
+                p_int = polyint(poly(diagm(-[0:j - 1; j + 1:s - 1])))
+                b[s, s - j] = ((-1)^j / factorial(j)
+                               / factorial(s - 1 - j) * polyval(p_int, 1))
+            end
+        end
+    end
+
+    # TODO: use a better data structure here (should be an order-element circ buffer)
+    dy = similar(y)
+    for i = 1:length(tspan)-1
+        # Need to run the first several steps at reduced order
+        steporder = min(i, order)
+        dy[i] = F(tspan[i], y[i])
+
+        # Check step size is equal or very near equal
+        if !isapprox(dt[i],dt[1])
+            error("Step sizes are not of constant length")
+        end
+
+        y[i+1] = y[i]
+        for j = 1:steporder
+            y[i+1] += dt[i]*b[steporder, j]*dy[i-(steporder-1) + (j-1)]
+        end
+    end
+    return vcat(tspan), y
+end
+
+# Use order 4 by default
+ode4ms(F, y0, tspan) = ODE.ode_ms(F, y0, tspan, 4)
+ode5ms(F, y0, tspan) = ODE.ode_ms(F, y0, tspan, 5)
+
+
+"""
+    t,y = ode_am((odefun, y0, tspan, order=4) with tspan = [t0:dt:tfinal]
+
+Fixed-Equal Step Adams-Moulton PECE solver for nonstiff problems
+
+(Main ref: Hairer & Wanner 1996, Vol I, p.359-360, 372)
+
+#Note: this function works for both scalar-like and array-like input
+
+#Note: If method is run at order k, the first step is taken at order 1,
+and the order increases each step until it is order k on the kth step.
+Thus results may be of lower accuracy than expected.
+"""
+
+##Function: ode_am(F,y0, t, order(optional))
+##order is set to 4 by default
+function ode_am(F,y0, t,order::Integer = 4)
+    if (1 <= order <= 4)
+        b_imp = am_imp_coefficients3
+        b_exp = ms_coefficients4
+    else
+        #calculating higher order coefficients for implicit Adams Moulton method
```

Comment:
Should this part for determining more coefficients instead be a separate function?